src/cpp/ext/otel/otel_server_call_tracer.cc

```diff
@@ -56,7 +56,14 @@ namespace {
 
 class OpenTelemetryServerCallTracer : public grpc_core::ServerCallTracer {
  public:
-  OpenTelemetryServerCallTracer() : start_time_(absl::Now()) {}
+  explicit OpenTelemetryServerCallTracer(const grpc_core::ChannelArgs& args)
+      : start_time_(absl::Now()) {
+    for (const auto& plugin_option :
```

Comment:
Instead of duplicating all of this code to determine the active set of options and then iterate over them, I wonder if it could be refactored into a common view object.  I'm thinking something like this:

```
class ActivePluginOptionsView {
 public:
  explicit ActivePluginOptionsView(absl::string_view target) {
    for (size_t i = 0; i < OpenTelemetryPluginState().plugin_options.size(); ++i) {
      const auto& plugin_option = OpenTelemetryPluginState().plugin_options[i];
      if (plugin_option != nullptr &&
          plugin_option->IsActiveOnClientChannel(target)) {
        active_mask_.set(i);
      }
    }
  }

  void ForEach(absl::FunctionRef<void(const OpenTelemetryPluginOption&)> func) const {
    for (size_t i = 0; i < OpenTelemetryPluginState().plugin_options.size(); ++i) {
      const auto& plugin_option = OpenTelemetryPluginState().plugin_options[i];
      if (active_mask_.is_set(i)) func(*plugin_option);
    }
  }

 private:
  BitSet<64> active_mask_;
};
```

You can instantiate that object in both the server and client code, and then you can call it using something like this:

```
active_plugin_options_view_.ForEach(
    [&](const OpenTelemetryPluginOption& plugin_option) {
      auto* labels_injector = plugin_option.labels_injector();
      if (labels_injector != nullptr) {
        labels_injector->AddLabels(send_initial_metadata, injected_labels_from_plugin_options_[i].get());
      }
    });
```

Reply:
Suggestion taken with some modifications.. I think the total number of lines is higher than before but I like it since we factored out the common code